### PR TITLE
[SPARK-20889][SparkR] Grouped documentation for NONAGGREGATE column methods

### PR DESCRIPTION
**Pull request by actuaryzhang**
## What changes were proposed in this pull request?

Grouped documentation for nonaggregate column methods.

### CODE REVIEW

**Comment by actuaryzhang** [comment]
jenkins, retest this please

**Comment by actuaryzhang** [comment]
jenkins, retest this please

**Comment by actuaryzhang** [comment]
jenkins, retest this please

**Comment by actuaryzhang** [comment]
jenkins, retest this please

**Comment by actuaryzhang** [comment]
jenkins, retest this please

**Comment by felixcheung** [comment]
jenkins, retest this please

**Comment by actuaryzhang** [comment]
jenkins, retest this please

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -132,23 +132,40 @@ NULL
...
-#' @rdname lit
-#' @name lit
+#' @examples
+#' \dontrun{
+#' # Dataframe used throughout this doc
+#' df <- createDataFrame(cbind(model = rownames(mtcars), mtcars))}
+NULL
+
+#' @details
+#' \code{lit}: A new \linkS4class{Column} is created to represent the literal value.
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: this format is actually kinda weird. let's fix it? I don't think we need to link to Column

File: R/pkg/R/functions.R [hunk 2]
```diff
@@ -3554,21 +3493,17 @@ setMethod("grouping_id",
...
+#' @aliases input_file_name input_file_name,missing-method
 #' @export
 #' @examples
-#' \dontrun{
-#' df <- read.text("README.md")
 #'
-#' head(select(df, input_file_name()))
-#' }
+#' \dontrun{
+#' tmp <- read.text("README.md")
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: why rename to `tmp` though?

File: R/pkg/R/functions.R [hunk 3]
```diff
@@ -824,32 +835,23 @@ setMethod("initcap",
             column(jc)
           })
 
-#' is.nan
-#'
-#' Return true if the column is NaN, alias for \link{isnan}
-#'
-#' @param x Column to compute on.
+#' @details
+#' \code{is.nan}: Alias for \link{isnan}.
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: roxygen does this by text order, I think - doesn't it make this go first, before isnan? perhaps we swap the order of code?

File: R/pkg/R/functions.R [hunk 4]
```diff
@@ -132,23 +132,40 @@ NULL
...
+#'
 #' \dontrun{
-#' lit(df$name)
-#' select(df, lit("x"))
-#' select(df, lit("2015-01-01"))
-#'}
+#' tmp <- mutate(df, v1 = lit(df$mpg), v2 = lit("x"), v3 = lit("2015-01-01"),
+#'                   v4 = negate(df$mpg), v5 = expr('length(model)'),
+#'                   v6 = greatest(df$vs, df$am), v7 = least(df$vs, df$am),
+#'                   v8 = column("mpg"))
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: is there example for 
```
nanvl(df$c, x)
coalesce(df$c, df$d, df$e) 
```

that I've missed?

File: R/pkg/R/functions.R [hunk 5]
```diff
@@ -132,23 +132,40 @@ NULL
...
 #'
-#' A new \linkS4class{Column} is created to represent the literal value.
-#' If the parameter is a \linkS4class{Column}, it is returned unchanged.
+#' Non-aggregate functions defined for \code{Column}.
 #'
-#' @param x a literal value or a Column.
+#' @param x Column to compute on. In \code{lit}, it is a literal value or a Column.
+#'          In \code{monotonically_increasing_id}, it should be empty.
+#' @param y Column to compute on.
+#' @param ... additional argument(s). In \code{expr}, it contains an expression character
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: `In \code{expr}, it contains an expression character` - this isn't quite right actually - it's in `x` in expr, not as ... parameter

File: R/pkg/R/functions.R [hunk 6]
```diff
@@ -132,23 +132,40 @@ NULL
...
-#' lit
+#' Non-aggregate functions for Column operations
 #'
-#' A new \linkS4class{Column} is created to represent the literal value.
-#' If the parameter is a \linkS4class{Column}, it is returned unchanged.
+#' Non-aggregate functions defined for \code{Column}.
 #'
-#' @param x a literal value or a Column.
+#' @param x Column to compute on. In \code{lit}, it is a literal value or a Column.
+#'          In \code{monotonically_increasing_id}, it should be empty.
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: and same for `input_file_name` - btw, should be empty might be a bit confusing? how about `In ......., Should be used with no argument.`?

File: R/pkg/R/functions.R (hunk 5 above)
Comment (felixcheung) [inline R/pkg/R/functions.R]: and so in all other cases in this group, `...` is expected for other columns. perhaps we can say `additional Columns`

File: R/pkg/R/functions.R (hunk 6 above)
Comment (actuaryzhang) [inline R/pkg/R/functions.R]: Yes, I was just copying from the old doc. 
I now remove this and add `The method should be used with no argument.` to the two individual methods.

File: R/pkg/R/functions.R (hunk 5 above)
Comment (actuaryzhang) [inline R/pkg/R/functions.R]: Right, thanks for catching this.

File: R/pkg/R/functions.R (hunk 1 above)
Comment (actuaryzhang) [inline R/pkg/R/functions.R]: updated.

File: R/pkg/R/functions.R [hunk 7]
```diff
@@ -2819,20 +2775,26 @@ setMethod("unix_timestamp", signature(x = "Column", format = "character"),
...
+#' @examples
+#'
+#' \dontrun{
+#' tmp <- mutate(df, mpg_na = otherwise(when(df$mpg > 20, df$mpg), lit(NaN)),
+#'                   mpg2 = ifelse(df$mpg > 20 & df$am > 0, 0, 1),
+#'                   mpg3 = ifelse(df$mpg > 20, df$mpg, 20.0))
+#' head(tmp)
+#' tmp <- mutate(tmp, ind_na1 = is.nan(tmp$mpg_na), ind_na2 = isnan(tmp$mpg_na))
+#' head(select(tmp, coalesce(tmp$mpg_na, tmp$mpg)))
+#' head(select(tmp, nanvl(tmp$mpg_na, tmp$hp)))}
```
Comment (actuaryzhang) [inline R/pkg/R/functions.R]: @felixcheung  Examples for `coalesce` and `nanvl` are here.

File: R/pkg/R/functions.R (hunk 4 above)
Comment (actuaryzhang) [inline R/pkg/R/functions.R]: See L2796.

File: R/pkg/R/functions.R (hunk 3 above)
Comment (actuaryzhang) [inline R/pkg/R/functions.R]: OK, swapped the order.

File: R/pkg/R/functions.R (hunk 2 above)
Comment (actuaryzhang) [inline R/pkg/R/functions.R]: To avoid overwriting the dataframe example `df` used throughout the doc.

**Review by felixcheung — APPROVED** [review]
LGTM.

**Comment by felixcheung** [comment]
merged to master

**Pull request closed without merging**